pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/MailboxReceiveNode.java

```diff
@@ -57,6 +57,8 @@ public MailboxReceiveNode(int stageId, DataSchema dataSchema, int senderStageId,
   }
 
   public int getSenderStageId() {
+    assert _sender == null || _sender.getStageId() == _senderStageId
```

Comment:
Is it better to assert in `setSender` then there will be a trace of where it is set to null.

Reply:
I may be wrong, but in `setSender` there is now way this may fail (given we set both). This is just more a reminder of the invariant for readers.

Reply:
Oh - misread the statement. I was thinking `!=`. nvm